whylog_vim/proxy/log_reader/__init__.py

```diff
@@ -1,24 +1,41 @@
+import six
+from whylog.log_reader.exceptions import NoLogTypeError
+from whylog_vim.consts import EditorStates, Messages
 from whylog_vim.output_formater.log_reader_formater import LogReaderOutput
 
 
 class LogReaderProxy(object):
-    def __init__(self, editor, log_reader):
-        self.editor = editor
+    def __init__(self, log_reader, config, editor, main_proxy):
         self._log_reader = log_reader
+        self._config = config
+        self._editor = editor
+        self._main_proxy = main_proxy
+
+    def _ask_log_type(self, log_types):
+        pass
 
     def new_query(self):
-        front_input = self.editor.get_front_input()
-        query_output = self._log_reader.get_causes(front_input)
-        self.output = LogReaderOutput.format_query(front_input, query_output)
-        self.output.set_default_callback_function(self.editor.close_query_window)
-        self.editor.create_query_window(self.output.get_content())
-        self.editor.go_to_query_window()
+        front_input = self._editor.get_front_input()
+        try:
+            query_output = self._log_reader.get_causes(front_input)
+        except NoLogTypeError:
+            log_types = [l for l in self._config.get_all_log_types()]
+            if log_types:
+                self._ask_log_type(log_types)
+            else:
+                six.print_(Messages.EMPTY_DATA_BASE)
```

Comment:
When user get NoLogTypeError I check if there are logTypes in config. If there are no log types it means that there are no parsers and rules too so database is empty.